src/core.js

```diff
@@ -64,7 +65,8 @@ $.extend( $.fn, {
 							.appendTo( validator.currentForm );
 					}
 
-					if ( validator.settings.submitHandler ) {
+					// Only call the submit handler in case the validator is not on debug mode
```

Comment:
Yeah, you're right! It doesn't have any additional value. I don't know why I wrote that comment in the first place (it was around 2AM when I created this PR). Will remove it in a moment.

As for the console.log, it will not make sense to print that the submitHandler will not be called when it supposed to not be called in the first place. So, we will just let it that way.